let caller method render flash message for sublist screens.

h-kataria opened the pull request:
- Fixed respective 'button' methods to render flash message partial when task(clone,migrate,publish) are not supported for selected items when user trees to perform these tasks from list of VMs thru relationships. render_flash_not_applicable_to_model will only render flash message when performing tasks directly from VM explorers.
- Added spec test to verify fix.

https://bugzilla.redhat.com/show_bug.cgi?id=1263326

@martinpovolny @dclarizio please review and suggest if there is another way this issue can be fixed. To recreate the issue Go to Infrastructure Providers, Go to SCVMM Provider summary screen, from there Click on VMs in the relationships box, Select one or more VMs from the list and click on Migrate selected Items button under Lifecycle button. You should see double render error in the log, you should be able to see this behavior for any of the CIs that have link to view VMs in the relationship box on their respective summary screens. Selected VM has to be the one that do not support Migrate task, which is why i used VMs under a SCVMM provider. Let me know if you have questions or need help recreating this issue.

h-kataria commented:
@martinpovolny please review.

martinpovolny commented on app/controllers/cim_instance_controller.rb:
```diff
@@ -92,6 +92,8 @@ def process_button
       end
     end
 
+    return if performed? # return if another method has already redirected or rendered
```
@h-kataria : I'd say that the comment is not needed here (and on the other places)

martinpovolny commented on spec/controllers/ems_common_controller_spec.rb:
```diff
@@ -130,5 +130,36 @@
+
+      it "when VM Migrate is pressed for unsupported type" do
+        controller.stub(:role_allows).and_return(true)
+        controller.stub(:performed?).and_return(true)
```
@h-kataria : should we stub this? I'd say that we are testing that we have the render logic ok and `performed?` is part of the logic. But it my be just that I don't get something here ;-)

h-kataria replied:
@martinpovolny when running spec test for some reason performed? was always returning false. Maybe i am doing something incorrectly, i will look again.

h-kataria replied:
@martinpovolny i looked at it again, performed? seems to be returning false when i run from spec test, even tho it is rendering flash message.

martinpovolny commented:
@h-kataria : tested, looks good :+1: Please, see the 2 minor comments in-line.

matthewd commented:
I don't like this; if "just silently ignore a second attempt to render" was the reasonable thing to do, we'd be doing it upstream.

If a controller is so confused that it's trying to render twice, that is a bug in the action's control flow, which should be fixed.

Where is the first render occurring here, and why are we not returning at that point?

h-kataria commented:
@matthewd i didn't feel comfortable adding this fix to begin with, i have another approach that i am going to try to implement, hopefully that will be a better one.

h-kataria commented:
@matthewd @martinpovolny @dclarizio tried a different approach please review.

martinpovolny commented:
@matthewd : You are right. The approach is wrong. But most of the controller code has this issue. So to address the cause we would have to do a major refactoring. Of course, we want to do that and we try to do that as part of our work.

But here we need to make a small enough fix to just fix the bug. Refactoring should be done in a separate PR.

chessbyte commented:
@matthewd @martinpovolny @dclarizio Please review with rework from @h-kataria

martinpovolny commented:
@h-kataria : stype issues:
  1. you have `if/unless` modifiers with the same condition for 2 consequent statements  ---> normal `if/else/end` would look better for sure
  1. several of the modified code blocks look exactly the same, can we extract a method, please?

h-kataria commented:
@martinpovolny please re-review, addressed your comments, the only controller that is not using the new common method is host controller as it passes in additional parameters on a redirect.
